### ISSUE TITLE
[1.2 RC] registerCompiler: "bare" option from api.addFiles should be honored if not overridden by build plugin

### ISSUE THREAD

**Issue by socketwiz** [meteor/meteor#4817]
I'm trying to use react with meteor and so far everything is working great.  However according to the documentation if you add a file in the client/compatibility directory variables are not supposed to be wrapped in a new variable scope:

http://docs.meteor.com/#/full/structuringyourapp

    Files in this directory are executed without being wrapped in a new variable scope.

This seems to only work for files ending in .js. Are there any work-arounds for getting this same behavior with .jsx files too?

In order to reproduce create the following file {project}/client/compatibility/foo.js:

    var foo = {};

If you run your project in the browser and look at the source for foo.js it should be correct:

    var foo = {};

Now rename foo.js to foo.jsx, and this time it will be scoped:

    (function(){var foo = {};
    
    })();

How can I make a react variable that can be used in other files?

**Comment by stubailo** [comment]
Confirmed, here is a repro: https://github.com/stubailo/meteor-issue-4817

However, if what you want to do is create a React variable that can be used in other files, `client/compatibility` is _not_ the way to do it. In Meteor, you can create a app-global variable by simply omitting the `var` in the declaration:

```js
// Available in other files
PublicComponent = React.createClass({...});

// Only available in this file
var PrivateComponent = React.createClass({ ... });
```

You can read more about it in the docs: http://docs.meteor.com/#/full/namespacing

**Comment by socketwiz** [comment]
Thanks @stubailo

**Comment by glasser** [comment]
I think the issue here is that the jsx package needs to see the `bare` option in the fileOptions inside its compiler, and pass it through to the addJavaScript call.   (The ecmascript package needs to do this too probably?) @dgreensp this probably should be done for 1.2, unless we explicitly want to prevent users from having bare jsx/es6 files.

**Comment by dgreensp** [comment]
The coffeescript package supports `bare`, so we should probably do it.  This is easy, right, just like one line?

**Comment by glasser** [comment]
@dgreensp believe so!

**Comment by glasser** [comment]
Moving this to https://github.com/meteor/react-packages/issues/86

**Comment by glasser** [comment]
Oops, @stubailo pointed out this: https://github.com/meteor/meteor/blob/ff783e9a12ffa04af6fd163843a563c9f4bbe8c1/tools/compiler.js#L710-L715

So the `registerSourceHandler` behavior was that `bare` is passed through from package.js by default, and can be overridden by the build plugin in its `addJavaScript` call. We should probably make `registerCompiler`'s version do the same thing.  @dgreensp probably a 1.2 blocker?

**Comment by stubailo** [comment]
Fixed on 1.2 branch: https://github.com/meteor/meteor/commit/217bad7f108d2236e0767f9f9c2b93f543d3e16e
